src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/ActivitiesResource.kt

```diff
@@ -477,6 +478,48 @@ class ActivitiesResource(
   ): List<FindSuspendedAllocationsResponse> =
     allocationService.findSuspendedAllocations(prisonId, excludeProgramCodes, courseActivityId)
 
+  @PreAuthorize("hasRole('ROLE_NOMIS_ACTIVITIES')")
+  @GetMapping("/allocations/missing-pay-bands")
```

Comment:
The main change is to add a new endpoint to be called from the activities migration preview and provide a list of allocated prisoners who don't have a valid pay band (but there is a pay rate available on the activity, e.g. not interested in Unemployed which doesn't pay anyway).